pkg/loki/modules.go

```diff
@@ -654,13 +653,26 @@ func (t *Loki) initQueryFrontendTripperware() (_ services.Service, err error) {
 	return services.NewIdleService(nil, nil), nil
 }
 
-func (t *Loki) cacheGenClient() (generationnumber.CacheGenClient, error) {
+func (t *Loki) cacheGenClient() (*generationnumber.GenNumberLoader, error) {
```

Comment:
This should probably just return the interface type `queryrangebase.CacheGenNumberLoader`